### PR TITLE
Fix/update training booked spot!!

### PR DESCRIPTION
**Pull request by ebuka-nwabueze**
- Added certificate download feature.
- Update the Account page for booked trainings.

### CODE REVIEW

File: src/pages/docs/training-certificate/[orderId]/[participantId].tsx [hunk 1]
```diff
@@ -0,0 +1,282 @@
...
+	const { user } = useUser();
+	const router = useRouter();
+	const pdfRef = useRef<HTMLDivElement>(null);
+
+	const [attendanceData, setAttendanceData] =
+		useState<TrainingAttendanceType | null>(null);
+	const orderId = router.query.orderId as string;
+	const participantId = router.query.participantId as string;
+
+	useEffect(() => {
```
Comment (Ozoemenamsolomon) [inline src/pages/docs/training-certificate/[orderId]/[participantId].tsx]: do this in the `getServerProps` function. That's the benefit of using next js over ordinary react

File: src/pages/docs/training-certificate/[orderId]/[participantId].tsx [hunk 2]
```diff
@@ -0,0 +1,282 @@
...
+				imgX,
+				imgY,
+				imgWidth * ratio,
+				imgHeight * ratio,
+			);
+			pdf.save('certificate.pdf');
+		});
+	};
+
+	if (!user)
```
Comment (Ozoemenamsolomon) [inline src/pages/docs/training-certificate/[orderId]/[participantId].tsx]: look at other authenticated routes, we use `withPageAuthRequired` provided by auth0 to make sure only authenticated users get to the page at all

File: src/pages/api/update-training-bookedspot.ts
Comment (Ozoemenamsolomon) [inline src/pages/api/update-training-bookedspot.ts:1]: just create a `training-spot` route and use the REST methods respectively: in this case, use `PUT`

File: src/pages/api/download-training-certificate.ts
Comment (Ozoemenamsolomon) [inline src/pages/api/download-training-certificate.ts:1]: the route name could be improved, the route is just providing data for the certificate that would be processed in the frontend

File: src/pages/docs/training-certificate/[orderId]/[participantId].tsx (hunk 1 above)
Comment (ebuka-nwabueze) [reply src/pages/docs/training-certificate/[orderId]/[participantId].tsx]: Done ✅✅

File: src/pages/docs/training-certificate/[orderId]/[participantId].tsx (hunk 2 above)
Comment (ebuka-nwabueze) [reply src/pages/docs/training-certificate/[orderId]/[participantId].tsx]: Done ✅✅

File: src/pages/api/download-training-certificate.ts
Comment (ebuka-nwabueze) [reply src/pages/api/download-training-certificate.ts:1]: Deleted the api route due to accessing the database directly from getServerSideProps.

File: src/pages/api/update-training-bookedspot.ts
Comment (ebuka-nwabueze) [reply src/pages/api/update-training-bookedspot.ts:1]: Done ✅✅

**Pull request merged**